iac-extensions/kubernetes/src/test/resources/checks/DeprecatedCodeCheck/helm/helm3/templates/deprecated_code.yaml

```diff
@@ -0,0 +1,56 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-configmap
+data:
+  myvalue: "Hello World"
+  {{- /*Noncompliant@+1 {{"Capabilities.KubeVersion.GitVersion" is deprecated since Helm 3, use "Capabilities.KubeVersion.Version" instead.}}*/}}
+  kubeVersionDeprecated: {{ .Capabilities.KubeVersion.GitVersion }}
+  {{- /*                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ */}}
+  # Noncompliant@+1
+  kubeVersionDeprecatedUsedInMoreComplicatedSetup: {{ .Capabilities.KubeVersion.GitVersion | quote }}
+  kubeVersion: {{ .Capabilities.KubeVersion.Version }}
+  kubeVersionUsedInMoreComplicatedSetup: {{ .Capabilities.KubeVersion.Version | quote }}
+
+---
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: "name"
+spec:
+  # Noncompliant@+1
+  {{- if semverCompare ">=1.18-0" .Capabilities.KubeVersion.GitVersion }}
+  ingressClassName: {{ .Values.ingress.className |  default "nginx" }}
+  {{- end }}
+  # Noncompliant@+1
+  {{- if and .Values.ingress.className (not (semverCompare ">=1.18-0" $.Capabilities.KubeVersion.GitVersion)) }}
+  ingressClassName: "test"
+  {{- end }}
+  # It's not possible to raise inside the usage of "Capabilities.KubeVersion.GitVersion" inside the .tpl file
```

Comment:
Do we have a ticket to address it? If not please create a new ticket.

Reply:
A similar ticket is https://sonarsource.atlassian.net/browse/SONARIAC-1309, however now that we can parse the Go AST, we can parse the `.tpl` files as well and execute checks on them. I created a ticket for that: https://sonarsource.atlassian.net/browse/SONARIAC-1616